components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/APIMgtGatewayConstants.java

```diff
@@ -90,5 +90,9 @@
     public static final String XSD_URL = "xsdURL";
     public static final String UTF8 = "UTF-8";
 
+    /**
+     * Web socket header for jwt assertion.
+     * */
+    public static final String WS_JWT_TOKEN_HEADER = "websocket.custom.header.X-JWT-Assertion";
```

Comment:
Why can't we use the same standard header name "X-JWT-Assertion"?

Reply:
Hi Fazlan,

The headers in the client handshake are not sent to the backend directly. In the WebSocketTransportSender, the headers are added to the axis2 message context as properties. There we cannot filter the headers that should be sent to the backend. So, we prefix the actual header that should be sent to the backend and add it as a header.

In the TransportSender, the headers are filtered by the prefix, extract the actual header and attach it to the handshake request. 

So in this scenario, the actual header that is sent to the backend is X-JWT-Assertion.